lib/admin-page.php

```diff
@@ -304,33 +304,59 @@ function classicpress_check_can_migrate() {
 	);
 	echo '<table id="cp-preflight-checks">' . "\n";
 
+	$disable_wp_version_check = apply_filters( 'classicpress_ignore_wp_version', false );
+
 	// Check: Supported WP version
 	// More versions can be added if they pass the plugin's automated tests.
-	global $wp_version;
-	$wp_version_min = '4.9.0';
-	$wp_version_max = '4.9.8';
-	if (
-		// Version is outside of our "stable release" range...
-		(
-			version_compare( $wp_version, $wp_version_min, 'lt' ) ||
-			version_compare( $wp_version, $wp_version_max, 'gt' )
-		) &&
-		// ... and it's not a known development release
-		! preg_match( '#^4\.9\.9-(alpha|beta)\b#', $wp_version )
-	) {
-		$preflight_checks['wp_version'] = false;
-		echo "<tr>\n<td>$icon_preflight_fail</td>\n<td>\n";
-	} else {
+	
+	if( !empty( $disable_wp_version_check ) ) { // Skip the test and issue a warning
+		$preflight_check_wp_version_status = $icon_preflight_warn;
 		$preflight_checks['wp_version'] = true;
-		echo "<tr>\n<td>$icon_preflight_pass</td>\n<td>\n";
+		
+	} else { // Run the check
+		global $wp_version;
+		
+		$wp_version_min = '4.9.0';
+		$wp_version_max = '4.9.8';
+		
+		// Default values - skips one else
+		$preflight_check_wp_version_status = $icon_preflight_pass;
+		$preflight_checks['wp_version'] = true;
+		
+		// Version is outside of our "stable release" range...
+		if ( 
+			( 
+				version_compare( $wp_version, $wp_version_min, 'lt' ) ||
+				version_compare( $wp_version, $wp_version_max, 'gt' )
+			) &&
+			// ... and it's not a known development release
+			! preg_match( '#^4\.9\.9-(alpha|beta)\b#', $wp_version )
+		) {
+			$preflight_checks['wp_version'] = false;
+			echo "<tr>\n<td>$icon_preflight_fail</td>\n<td>\n";
+		}
+		
 	}
+		
+	echo "<tr>\n<td>$preflight_check_wp_version_status</td>\n<td>\n";
+	
 	echo "<p>\n";
+	
+	// Warn the user thoroughly: From here there be dragons!
+	if( !empty( $disable_wp_version_check ) ) {
+		_e( 
+			'<strong>Please be aware:</strong> The preflight check for supported WordPress versions has been <strong>disabled</strong>. Without this, we cannot guarantee that the migration process is going to work, and may even leave your current installation partially broken. Proceed on your own risk!', 
+			'switch-to-classicpress' 
+		);
+	}
+	
 	printf( __(
 		/* translators: 1: minimum supported WordPress version, 2: maximum supported WordPress version */
 		'This plugin supports WordPress versions <strong>%1$s</strong> to <strong>%2$s</strong> (and some newer development versions).',
 		'switch-to-classicpress'
 	), $wp_version_min, $wp_version_max );
-	echo "<br>\n";
+	
+	echo "<br />\n";
```

Comment:
I don't have a problem with this change. ([It's a mess, but either way is valid HTML5](https://stackoverflow.com/questions/1946426/html-5-is-it-br-br-or-br).)

However, it would be good to make this change in a separate PR and update all of these tags at once.

Reply:
Personally I'd totally rewrite the plugin with proper OOP, seperate templates, CSS et all, but I do understand the need for a basic, plain code setup.

Reply:
That has been proposed before (cc @glagonikas).  I think it is not urgent but it is probably worth doing in a separate PR.  I'd use a class for each check, and methods with standardized return values to indicate their status.

I also expect we will want to re-use this code structure for upgrades between ClassicPress versions, so it would be nice to keep it more maintainable.